### DISCUSSION TITLE
Form with integration over two of the three dimensions

### DISCUSSION THREAD

**Discussion by cjham** [firedrakeproject/firedrake#3441] [Firedrake support] [answered]
I have a potential form for which I only want to integrate over two of the dimensions rather than all three. The domain is a torus and I want to integrate over the minor radius and the poloidal angle (short way round) but not the toroidal angle (long way round). Is this possible? 
If I understood correctly this sort of thing could be possible if the mesh is extruded? Could I rotate a disk mesh around an axis to get a torus mesh via extrusion? And then just integrate over the minor radius and poloidal angle? Hopefully this question makes sense. Please let me know if you need more details.

**Comment by colinjcotter** [comment, marked as answer]
You can make a finite element space that is constant in the extruded direction (see finite element spaces on extruded meshes section in the manual). To get what you want you would need to extrude a toroidal surface mesh with a custom extrusion option in ExtrudedMesh.
